onnxruntime/python/tools/transformers/fusion_attention.py

```diff
@@ -693,12 +692,13 @@ def create_multihead_attention_node(
             mha_inputs.append("")
 
         # Add optional inputs for MHA
-        if past_k and past_v and past_k in graph_input_names and past_v in graph_input_names:
+
+        if past_k and past_v:
```

Comment:
if past_k and past_v:
    mha_inputs.extend([key_padding_mask, add_qk, past_k, past_v])
elif key_padding_mask or add_qk:
    mha_inputs.extend([key_padding_mask, add_qk])